File: app/channels/dm_room_channel.rb
class DmRoomChannel < ApplicationCable::Channel
  def subscribed
    first_id, second_id = [current_user.id, params["receive_user_id"]].sort
    stream_from "dm_room_channel_#{first_id}_#{second_id}"
  end

  def unsubscribed
  end

  def speak(data)
    Dm.create! content: data["dm"], send_user_id: current_user.id, receive_user_id: params["receive_user_id"]
  end
end

File: app/javascript/channels/dm_room_channel.js
import consumer from "./consumer"

$(function () {
  const dmChannel = consumer.subscriptions.create({
    channel: 'DmRoomChannel',
    receive_user_id: $('#dms').data('receive_user_id')
  }, {
    connected() {
      // Called when the subscription is ready for use on the server
    },

    disconnected() {
      // Called when the subscription has been terminated by the server
    },

    received: function (data) {
      return $('#dms').append(data['dm']);
    },

    speak: function (dm) {
      return this.perform('speak', {
        dm: dm
      });
    }
  });

  $(document).on('keypress', '[data-behavior~=dm_room_speaker]', function (event) {
    if (event.keyCode === 13) {
      dmChannel.speak(event.target.value);
      event.target.value = '';
      return event.preventDefault();
    }
  });
});